src/trunk.c

```diff
@@ -4198,12 +4211,16 @@ trunk_flush_fullest(trunk_handle *spl, page_handle *node)
       tid = platform_get_tid();
    }
    if (trunk_pivot_needs_flush(spl, node, fullest_pivot_data)) {
-      trunk_flush(spl, node, fullest_pivot_data, FALSE);
+      bool flushed = trunk_flush(spl, node, fullest_pivot_data, FALSE);
+      if (!flushed) {
```

Comment:
Again, the question arises as to what tests ran into this bug? Presumably a stress test?

What can we do to integrate that test / stress-test in a reduced form to cause this bug and, thus, verify that this fix does fix that issue? Overall, test stability will improve if we can  bring online some test case with this fix.

Reply:
I should note that this didn't trigger an outwardly visible bug, but rather I observed a bunch of looping at L3353 in the debugger. In some sense that's worse. This was the same perf test as above.

Without tests that have access to trunk node structure, which is a big undertaking, I'm really not sure how we can directly test this sort of thing. I'll think about it.